src/test/txvalidationcache_tests.cpp

```diff
@@ -105,9 +105,13 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
 static void ValidateCheckInputsForAllFlags(const CTransaction &tx, uint32_t failing_flags, bool add_to_cache)
 {
     PrecomputedTransactionData txdata(tx);
-    // If we add many more flags, this loop can get too expensive, but we can
-    // rewrite in the future to randomly pick a set of flags to evaluate.
-    for (uint32_t test_flags=0; test_flags < (1U << 16); test_flags += 1) {
+
+    FastRandomContext insecure_rand(true);
+
+    for (uint32_t count=0; count < 10000; count++) {
+        // Randomly selects flag combinations
+        uint32_t test_flags = insecure_rand.randrange(HIGHEST_FLAG << 1);
```

Comment:
Make this implicit conversion from `uint64_t` to `uint32_t` explicit :-)

Reply:
Actually there's quite a lot of confusion of types around these types. `randrange` return a ` `uint64_t`, but `CheckInputs` accepts a `unsigned int`, while the flags are actually an enum which has an implementation-specific type. I guess the best is to change it here to `unsigned int` (and make the conversion explicit), and maybe open a separate PR for fixing these type inconsistencies.